rabbit-escape-engine/src/rabbitescape/engine/textworld/TextWorldManip.java

```diff
@@ -210,53 +212,110 @@ public static World createEmptyWorld( int width, int height )
 
     public static String[] renderCompleteWorld( World world, boolean meta )
     {
+        return renderCompleteWorld( world, meta, false );
+    }
+    
+    /**
+     * @param world           The World to render.
+     * @param meta            If true the metadata is included
+     * @param minimalistMeta  If true only the metadata that is useful in a rel file is added.
+     * @return                The lines as an array.
+     */
+    public static String[] renderCompleteWorld( World world, boolean meta, boolean minimalistMeta )
+    {
         Chars chars = new Chars( world, true );
 
         BlockRenderer.render( chars, world.blockTable );
         RabbitRenderer.render( chars, world.rabbits );
         ThingRenderer.render( chars, world.things );
 
-        String[] things = charsToComplete( chars );
+        String[] things = charsToComplete( chars, world.comments );
 
         if ( meta )
         {
-            return concat( metaLines( world ), things );
+            List<String> worldComments = new ArrayList<String>();
+            maybeInsertComment( Comment.WORLD_ASCII_ART, worldComments, world);
+            return concat( metaLines( world, minimalistMeta),
+                           worldComments.toArray( new String[]{} ),
+                           things );
         }
         else
         {
             return things;
         }
     }
+    
+    private static void maybeInsertComment( String key, List<String> ret, Comment[] comments )
+    {
+        for( Comment c: comments)
+        {
+            if( c.follows( key ))
+            {
+                ret.add( c.text );
+            }
+        }
+    }
+    
+    private static void maybeInsertComment( String key, List<String> ret, World w )
+    {
+        maybeInsertComment( key, ret, w.comments );
+    }
 
-    private static String[] metaLines( World world )
+    private static String[] metaLines( World world, boolean minimalist )
     {
         List<String> ret = new ArrayList<String>();
 
-        ret.add( metaLine( name,         world.name ) );
-        ret.add( metaLine( description,  world.description ) );
-        ret.add( metaLine( author_name,  world.author_name ) );
-        ret.add( metaLine( author_url,   world.author_url ) );
-        addMetaKeyArrayLines( ret, hint, world.hints );
-        addMetaKeyArrayLines( ret, solution, world.solutions );
-        ret.add( metaLine( num_rabbits,  world.num_rabbits ) );
-        ret.add( metaLine( num_to_save,  world.num_to_save ) );
-        ret.add( metaLine( rabbit_delay, world.rabbit_delay ) );
-        ret.add( metaLine( num_saved,    world.num_saved ) );
-        ret.add( metaLine( num_killed,   world.num_killed ) );
-        ret.add( metaLine( num_waiting,  world.num_waiting ) );
-        ret.add( metaLine( paused,       world.paused ) );
-
+        maybeInsertComment( name, ret,            world );
+        ret.add( metaLine(  name,             world.name ) );
```

Comment:
A function for `maybeInsertComment` + `ret.add( metaLine` would make this clearer.